src/main/java/org/folio/search/service/converter/preprocessor/InstanceEventPreProcessor.java

```diff
@@ -56,20 +59,54 @@ public List<ResourceEvent> preProcess(ResourceEvent event) {
     if (log.isDebugEnabled()) {
       log.debug("preProcess::Starting instance event pre-processing [{}]", event);
     }
-    if (startsWith(getResourceSource(event), SOURCE_CONSORTIUM_PREFIX)) {
+
+    List<ResourceEvent> events;
+
+    if (isUpdateForInstanceSharing(event)) {
+      events = prepareClassificationEventsOnInstanceSharing(event);
+    } else if (startsWith(getResourceSource(event), SOURCE_CONSORTIUM_PREFIX)) {
       log.info("preProcess::Finished instance event pre-processing. No additional events created for shadow instance.");
       return List.of(event);
+    } else {
+      events = prepareClassificationEvents(event);
     }
 
-    var events = prepareClassificationEvents(event);
-
     log.info("preProcess::Finished instance event pre-processing");
     if (log.isDebugEnabled()) {
       log.debug("preProcess::Finished instance event pre-processing. Events after: [{}], ", events);
     }
     return events;
   }
 
+  private boolean isUpdateForInstanceSharing(ResourceEvent event) {
+    return event.getType() == UPDATE
+      && startsWith(getResourceSource(getNewAsMap(event)), SOURCE_CONSORTIUM_PREFIX)
+      && SOURCE_FOLIO.equals(getResourceSource(getOldAsMap(event)));
```

Comment:
It could be not just FOLIO but other sources also. You may check that 
old.source == new.source - SOURCE_CONSORTIUM_PREFIX